File: leetcode/lc-us/cpp/1814.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int countNicePairs(vector<int> &nums) {
        unordered_map<int, int> cnt;
        for (int n: nums) {
            string s = to_string(n);
            reverse(s.begin(), s.end());
            int m = stoi(s);
            cnt[n - m]++;
        }
        long long res = 0, MOD = 1e9 + 7;
        for (auto[k, v]: cnt) {
            res = (res + (long) (v - 1) * v / 2) % MOD;
        }
        return (int) res;
    }
};

File: leetcode/lc-us/py/1814.py
from collections import defaultdict

from typing import List


class Solution:
    def countNicePairs(self, nums: List[int]) -> int:
        res = 0
        cnt = defaultdict(int)
        for num in nums:
            rev = int(str(num)[::-1])
            res += cnt[num - rev]
            res %= 10 ** 9 + 7
            cnt[num - rev] += 1
        return res
